### PR TITLE
Add tests for macOS

### PR DESCRIPTION
**Pull request by yatarkan**

### CODE REVIEW

File: .github/workflows/genai_package.yml [hunk 1]
```diff
@@ -31,6 +34,35 @@ jobs:
...
+      - run: curl https://storage.openvinotoolkit.org/repositories/openvino/packages/pre-release/2024.2.0rc1/macos/m_openvino_toolkit_macos_12_6_2024.2.0.dev20240524_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
+      - run: brew install coreutils ninja scons
+      - run: source ./ov/setupvars.sh && cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ./ -B ./build/
+      - run: source ./ov/setupvars.sh && cmake --build ./build/ --config ${{ matrix.build-type }} --target package -j
+      - run: source ./ov/setupvars.sh && cmake --install ./build/ --config ${{ matrix.build-type }} --prefix ov
+      - run: ov/samples/cpp/build_samples.sh -i ${{ github.workspace }}/s\ pace
+        if: ${{ 'Release' == matrix.build-type }}  # build_samples enforces Release build
+      - run: source ./ov/setupvars.sh && python -m pip install --upgrade-strategy eager -r text_generation/causal_lm/cpp/requirements.txt
+        if: ${{ 'Release' == matrix.build-type }}
+      - run: source ./ov/setupvars.sh && python -m pip install ./thirdparty/openvino_tokenizers/[transformers]
```
Comment (Wovchena) [inline .github/workflows/genai_package.yml]:
```suggestion
      - run: source ./ov/setupvars.sh && python -m pip install ./thirdparty/openvino_tokenizers/[transformers] --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release
```
A good strategy for you may be to review the new state of the file from scratch to catch such differences

File: .github/workflows/genai_python_lib.yml [hunk 2]
```diff
@@ -36,6 +39,38 @@ jobs:
...
+      - run: mkdir ./ov/
+      - run: curl https://storage.openvinotoolkit.org/repositories/openvino/packages/pre-release/2024.2.0rc1/macos/m_openvino_toolkit_macos_12_6_2024.2.0.dev20240524_x86_64.tgz | tar --directory ./ov/ --strip-components 1 -xz
+      - run: brew install coreutils ninja scons
+      - run: source ./ov/setupvars.sh && cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
+      - run: source ./ov/setupvars.sh && cmake --build ./build/ --config Release -j
+      # GitHub Actions already provides what is listed in ./requirements-build.txt but the internal
+      # build system doesn't. Install ./requirements-build.txt to detect possible conflicts.
+      - run: source ./ov/setupvars.sh && python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./requirements-build.txt --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release --verbose --verbose --verbose
+      - run: source ./ov/setupvars.sh && PYTHONPATH=./build/ python -c "from openvino_genai import LLMPipeline"
+      - run: source ./ov/setupvars.sh && python -m pip install . --config-settings=build-dir="build" --verbose --verbose --verbose
```
Comment (Wovchena) [inline .github/workflows/genai_python_lib.yml]:
```suggestion
      - run: source ./ov/setupvars.sh && python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./requirements-build.txt --extra-index-url https://storage.openvinotoolkit.org/simple/wheels/pre-release
      - run: source ./ov/setupvars.sh && PYTHONPATH=./build/ python -c "from openvino_genai import LLMPipeline"
      - run: source ./ov/setupvars.sh && python -m pip install . --config-settings=build-dir="build" --verbose
```
Excessive logging verbosity is causing lag in GitHub's web view. Reduce the verbosity for `openvino_tokenizers`, as failures from them are not anticipated. Retain only a single `--verbose` flag for `genai`.

**Review by Wovchena — CHANGES_REQUESTED** [review]

File: .github/workflows/genai_package.yml (hunk 1 above)
Comment (yatarkan) [reply .github/workflows/genai_package.yml]: Done

File: .github/workflows/genai_python_lib.yml (hunk 2 above)
Comment (yatarkan) [reply .github/workflows/genai_python_lib.yml]: Done

**Review by Wovchena — APPROVED** [review]

**Pull request merged**